src/spyglass/utils/mixins/ingestion.py

```diff
@@ -348,11 +353,36 @@ def validate_duplicates(self, entry_dict: Dict[dj.Table, List[dict]]):
         entry_dict : dict or Dict[dj.Table, List[dict]]
             The new entry or dict of table entries to validate against existing
             entries in the database.
+
+        Returns
+        -------
+        dict or Dict[dj.Table, List[dict]]
+            The new entries to insert after validation. Avoids need to flag
+            skip_duplicates
         """
+        entries_to_insert = dict()
         for table, table_entries in entry_dict.items():
-            for entry in self._adjust_keys_for_entry(table_entries):
+            if isinstance(table, type):
+                table = table()  # instantiate table object if class provided
+
+            entries_to_insert[table] = []
+            for table_entry in table_entries:
+                adjusted_entries = table._adjust_keys_for_entry([table_entry])
+                if not adjusted_entries:
+                    continue  # entry filtered out by adjustment
+                entry = adjusted_entries[0]
+                primary_entry = {
+                    k: v for k, v in entry.items() if k in table.primary_key
+                }
+                if not (table & primary_entry):
+                    # New entry, safe to insert
+                    entries_to_insert[table].append(table_entry)
+                    continue
+                # Existing entry, validate consistency
                 self.validate1_duplicate(table, entry)
 
+        return entries_to_insert
```

Comment:
is `_expected_duplicates`...
1. I expect this insert to contain things that may already exist in the table
2. I expect this file to contain multiple instances of the same insert into the table

If only the former, no changes needed. If the latter, we may want to remove dupes just before returning